### DISCUSSION TITLE
Turn NumLock on when the login screen shows up

### DISCUSSION THREAD

**Discussion by Mogaba** [FreeRDP/FreeRDP#8864] [Q&A]
I have a self-made "thin-client" where FreeRDP is started as a LightDM session. Our users want NumLock to be always turned on when RDP login screen shows up. I tried using setleds, numlockx and even xdotool to simulate NumLock pressing. All of them work fine with the default LightDM login screen but doesn't work with the FreeRDP session. Moreover, all these tools break NumLock when used with FreeRDP: they toggle the LED but don't change Numpad keys behaviour.

I'm not sure if it's FreeRDP issue but any help would be appreciated.

**Comment by akallabeth** [comment]
so you start a `RDP` session directly from the `LightDM`?
then issue some focus in/out to the started `xfreerdp` (I assume?) which triggers a local/remote keyboard state sync

**Reply by Mogaba** [reply]
> so you start a `RDP` session directly from the `LightDM`? then issue some focus in/out to the started `xfreerdp` (I assume?) which triggers a local/remote keyboard state sync

Sorry, I don't understand, what do you mean "focus in/out"?

**Reply by akallabeth** [reply]
you need to remove focus from the `xfreerdp` window (e.g. move your mouse out and back in) to force-trigger a sync of the state)
this can be done with something like `xdotool` automatically.

(the tools you try to change the state with just change the local keyboard state but fail to tell that to `xfreerdp` which needs to communicate the change to the remove windows server)
